### PR TITLE
Remove nested list in api result

### PR DESCRIPTION
**Pull request by Kelvin-M**
Companion PR:
https://github.com/SubstraFoundation/substra/pull/223
https://github.com/SubstraFoundation/substra-frontend/pull/81

### CODE REVIEW

**Review by jmorel — APPROVED** [review]

**Review by jmorel — COMMENTED** [review]
Tested through the companion PR in the frontend!

File: backend/substrapp/views/filters_utils.py [hunk 1]
```diff
@@ -100,6 +102,14 @@ def _get_model_tuple(model):
         raise NotImplementedError
 
 
+def flatten_without_duplicates(list_of_list):
```
Comment (Esadruhn) [inline backend/substrapp/views/filters_utils.py:105]: If speed is important, might be better to do it with a set:

```python
def flatten_set(list_of_list):
    res = set()
    for item in itertools.chain.from_iterable(list_of_list):
        if item not in res:
            res.add(item)
    return list(res)
```

```python
data = [[j for j in range(100)] for i in range(1000)]
%timeit flatten_without_duplicates(data)
%timeit flatten_set(data)
```
--> with list: 10 loops, best of 5: 241 ms per loop
--> with set: 100 loops, best of 5: 8.6 ms per loop

Comment (Kelvin-M) [reply backend/substrapp/views/filters_utils.py:105]: I don't know if it's a bottleneck of the application :)

**Review by Esadruhn — APPROVED** [review]
the comment on set is not blocking

File: backend/substrapp/views/filters_utils.py (hunk 1 above)
Comment (Esadruhn) [reply backend/substrapp/views/filters_utils.py:105]: Probably not ! Not blocking for the merge

**Review by inalgnu — APPROVED** [review]

**Pull request merged**